src/helpers/component-tree.ts

```diff
@@ -87,3 +87,34 @@ export function getHostSiblings(
     (sibling) => !hostSelves.includes(sibling)
   );
 }
+
+export function getCompositeParentOfType(
+  element: ReactTestInstance,
+  type: React.ComponentType
+) {
+  let current = element.parent;
+
+  while (!isHostElement(current)) {
+    // We're at the root of the tree
+    if (!current) {
+      return null;
+    }
+
+    if (current.type === type) {
+      return current;
+    }
+    current = current.parent;
+  }
+
+  return null;
+}
+
+export function isHostElementOfType(
```

Comment:
Seems like `for` should be more accurate then `of` here:

as in "this view is host element for type `Text` (exported from RN)", 

```suggestion
export function isHostElementForType(
```